cub/cub/block/block_run_length_decode.cuh

```diff
@@ -382,7 +382,13 @@ public:
     {
       decoded_items[i] = val;
       item_offsets[i]  = thread_decoded_offset - assigned_run_begin;
-      if (thread_decoded_offset == assigned_run_end - 1)
+      
+      // A thread only needs to fetch the next run if this was not the last loop iteration
+      // The cast to int32_t is needed to work around the compiler complaining about "pointless comparison of unsigned
+      // integer with zero" when DECODED_ITEMS_PER_THREAD is 1
+      const bool may_need_to_fetch_next_run =
+        static_cast<int32_t>(i) < (static_cast<int32_t>(DECODED_ITEMS_PER_THREAD) - 1);
+      if (may_need_to_fetch_next_run && (thread_decoded_offset == assigned_run_end - 1))
```

Comment:
Nitpick: I would generally prefer if we could separate the conditionals. We could also invert the condition to avoid the integer casting:
```suggestion
      const bool is_final_loop_iteration = i >= DECODED_ITEMS_PER_THREAD - 1;
      if (is_final_loop_iteration) {
          break;
      }
      
      if (thread_decoded_offset == (assigned_run_end - 1))
```

Reply:
Thanks, that's a great suggestion!